tests/components/minecraft_server/test_sensor.py

```diff
@@ -0,0 +1,362 @@
+"""Tests for Minecraft Server sensors."""
+from datetime import timedelta
+from unittest.mock import patch
+
+from mcstatus import BedrockServer, JavaServer
+from mcstatus.status_response import BedrockStatusResponse, JavaStatusResponse
+import pytest
+from syrupy import SnapshotAssertion
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    TEST_BEDROCK_STATUS_RESPONSE,
+    TEST_HOST,
+    TEST_JAVA_STATUS_RESPONSE,
+    TEST_PORT,
+)
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+@pytest.mark.parametrize(
+    ("mock_config_entry", "server", "status_response", "entity_id"),
+    [
+        (
+            "java_mock_config_entry",
+            JavaServer,
+            TEST_JAVA_STATUS_RESPONSE,
+            "sensor.minecraft_server_latency",
+        ),
+        (
+            "java_mock_config_entry",
+            JavaServer,
+            TEST_JAVA_STATUS_RESPONSE,
+            "sensor.minecraft_server_players_online",
+        ),
+        (
+            "java_mock_config_entry",
+            JavaServer,
+            TEST_JAVA_STATUS_RESPONSE,
+            "sensor.minecraft_server_world_message",
+        ),
+        (
+            "java_mock_config_entry",
+            JavaServer,
+            TEST_JAVA_STATUS_RESPONSE,
+            "sensor.minecraft_server_version",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_latency",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_players_online",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_world_message",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_version",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_map_name",
+        ),
+        (
+            "bedrock_mock_config_entry",
+            BedrockServer,
+            TEST_BEDROCK_STATUS_RESPONSE,
+            "sensor.minecraft_server_game_mode",
+        ),
+    ],
+)
+async def test_sensor(
```

Comment:
Maybe you want to checkout

https://github.com/home-assistant/core/blob/6f845497846c3463d151124555921d9d14b3cc11/tests/components/withings/test_sensor.py#L24-L41

This fetches it from the entity registry. This makes it that you only have to run snapshots in the future when introducing new entities.

(Note: not a must, just a suggestion :) )

Reply:
I think this approach would be best. Create a separate function which just returns the entity_ids present from the intergation and then you can check them

Reply:
With that I see the advantage, that I can get all sensor entities via the entity registry, and test them against the snapshot.  And even if I add entities in the future I would only need to update the snapshot and the test case would automatically cover the new entitites.

But then I have the disadvantage, that I don't test against an explicit list of expected sensor entities, or do I?
And why would I need a function which returns the sensor entitites? I didn't get that one, sorry.

Reply:
More like, currently you're having 3 parameterized tests with quite a few test cases. And I think you can break that down to just 3 tests with a for loop (or 3 parameterized tests with 2 cases - bedrock and java).

I'm noticing I'm rambling a little bit so let me give an example:

Create the tests with 2 parameterized test cases, bedrock and java (for each test). Then create or a static list of sensors to check (for each server type and pass them in) or create a function that just gets all entities from the registry. And then just run a for loop in the test to make sure the state of those entities are what you expect.

Does this make it clearer?

Reply:
Ah okay so you actually proposed two different approaches: One with the entity registry and one with passing the entities via a function and not via parametrize. I would prefer the 2nd option, as this tests against an explicit list of expected sensor entities. 
And the big advantage is that we reduce the number of test cases and also get rid of the huge parametrize blocks, I see 👍

Reply:
I reworked it, but instead of a function I'm using constant lists. Please have a look if it suits your proposal.

Reply:
One thing I'm wondering, why don't we test the value of the disabled entities?

Reply:
You mean to somehow enable the by default disabled entitites and test their values as well?

Reply:
In that piece of withings I linked I use a fixture that automatically enables disabled entities. This way you can also test the state of them.

Reply:
Nice one, done 👍